platform/security/filter/security-filter-authorization/src/main/java/org/codice/ddf/security/filter/authorization/AuthorizationFilter.java

```diff
@@ -94,19 +94,16 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 if (StringUtils.isEmpty(path)) {
                     path = httpRequest.getRequestURI();
                 }
-                ActionPermission actionPermission = new ActionPermission(path);
+
                 ContextPolicy policy = contextPolicyManager.getContextPolicy(path);
 
-                if (policy != null) {
+                if (policy != null && subject != null) {
                     Collection<CollectionPermission> permissions = policy
                             .getAllowedAttributePermissions();
-                    if (!CollectionUtils.isEmpty(permissions)) {
-                        permissions.add(new CollectionPermission(actionPermission));
-                        for (CollectionPermission permission : permissions) {
-                            if (subject == null || !subject
-                                    .isPermittedAll(permission.getPermissionList())) {
-                                permitted = false;
-                            }
+                    for (CollectionPermission permission : permissions) {
```

Comment:
We shouldn't need to loop through the permissions anymore. We should just be calling subject.isPermitted(keyValueCollectionPermission). I expected to see us update policy.getAllowedAttributePermissions() to return a KeyValueCollectionPermission. Then in this block you'd add the action to that KeyValueCollectionPermission and check the subject against that.